compiler/modules/types/string.bal

```diff
@@ -194,10 +50,12 @@ function createStringSubtype(boolean allowed, string[] values) returns SubtypeDa
 }
 
 final UniformTypeOps stringOps = {
-    union: stringSubtypeUnion,
+    union: function (SubtypeData d1, SubtypeData d2) returns SubtypeData =>
```

Comment:
Just put enumerableSubtypeUnion directly in the ops table (and make the parameter type be SubtypeData). Similarly for all the other functions in the ops table.